spec/AuthenticationAdapters.spec.js

```diff
@@ -2170,3 +2234,1055 @@ describe('facebook limited auth adapter', () => {
+    expect(modernAdapter2.validateSetUp).toHaveBeenCalledTimes(1);
+    const call2 = modernAdapter2.validateSetUp.calls.argsFor(0);
+    expect(call2[0]).toEqual({ id: 'modernAdapter2' });
+    expect(call2[1]).toEqual(modernAdapter2);
+    expect(call2[2].isChallenge).toBeUndefined();
+    expect(call2[2].master).toBeDefined();
+    expect(call2[2].object instanceof Parse.User).toBeTruthy();
+    expect(call2[2].original instanceof Parse.User).toBeTruthy();
+    expect(call2[2].user instanceof Parse.User).toBeTruthy();
+    expect(call2[2].original.id).toEqual(call2[2].object.id);
+    expect(call2[2].user.id).toEqual(call2[2].object.id);
+    expect(call2[2].object.id).toEqual(user.id);
+    expect(call2[3] instanceof Config).toBeTruthy();
+
+    const user2 = new Parse.User();
+    user2.id = user.id;
+    await user2.fetch({ useMasterKey: true });
+    expect(user2.get('authData')).toEqual({
+      modernAdapter: { id: 'modernAdapter' },
+      modernAdapter2: { id: 'modernAdapter2' },
+    });
+  });
+
+  it('should trigger correctly validateLogin', async () => {
+    spyOn(modernAdapter, 'validateSetUp').and.resolveTo({});
+    spyOn(modernAdapter, 'validateUpdate').and.resolveTo({});
+    spyOn(modernAdapter, 'validateLogin').and.resolveTo({});
+
+    await reconfigureServer({ auth: { modernAdapter }, allowExpiredAuthDataToken: false });
+    const user = new Parse.User();
+
+    // Signup
+    await user.save({ authData: { modernAdapter: { id: 'modernAdapter' } } });
+
+    expect(modernAdapter.validateSetUp).toHaveBeenCalledTimes(1);
+    // Login
+    const user2 = new Parse.User();
+    await user2.save({ authData: { modernAdapter: { id: 'modernAdapter' } } });
+
+    expect(modernAdapter.validateUpdate).toHaveBeenCalledTimes(0);
+    expect(modernAdapter.validateSetUp).toHaveBeenCalledTimes(1);
+    expect(modernAdapter.validateLogin).toHaveBeenCalledTimes(1);
+    const call = modernAdapter.validateLogin.calls.argsFor(0);
+    expect(call[0]).toEqual({ id: 'modernAdapter' });
+    expect(call[1]).toEqual(modernAdapter);
+    expect(call[2].object instanceof Parse.User).toBeTruthy();
+    expect(call[2].original instanceof Parse.User).toBeTruthy();
+    expect(call[2].isChallenge).toBeUndefined();
+    expect(call[2].master).toBeDefined();
+    expect(call[2].user).toBeUndefined();
+    expect(call[2].original.id).toEqual(user2.id);
+    expect(call[2].object.id).toEqual(user2.id);
+    expect(call[2].object.id).toEqual(user.id);
+    expect(call[3] instanceof Config).toBeTruthy();
+    expect(user2.getSessionToken()).toBeDefined();
+  });
+
+  it('should trigger correctly validateUpdate', async () => {
+    spyOn(modernAdapter, 'validateSetUp').and.resolveTo({});
+    spyOn(modernAdapter, 'validateUpdate').and.resolveTo({});
+    spyOn(modernAdapter, 'validateLogin').and.resolveTo({});
+
+    await reconfigureServer({ auth: { modernAdapter } });
+    const user = new Parse.User();
+
+    // Signup
```

Comment:
Remove this and the following single word comments

Reply:
please use github commit suggestion